src/NuGet.Core/NuGet.Protocol.Core.v3/Converters/SafeBoolConverter.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace NuGet.Protocol.Core.v3
+{
+    public class SafeBoolConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType) => objectType == typeof(bool);
+        public override bool CanRead { get { return true; } }
+        public override bool CanWrite { get { return false; } }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            switch (reader.TokenType)
+            {
+                case JsonToken.Null:
+                    return false;
+                case JsonToken.Boolean:
+                    return serializer.Deserialize<bool>(reader);
+                case JsonToken.String:
+                    return reader.Value.ToString().Equals("true", StringComparison.OrdinalIgnoreCase);
```

Comment:
Try this instead:
```c#
bool flag;
if (Boolean.TryParse(value.Trim(), out flag))
{
  return flag;
}

return false;
```

It is already [case-insensitive](https://msdn.microsoft.com/en-us/library/system.boolean.tryparse(v=vs.110).aspx).